bluezero/adapter.py

```diff
@@ -258,10 +259,13 @@ def _properties_changed(self, interface, changed, invalidated, path):
         Handle DBus PropertiesChanged signal and
         call appropriate user callback
         """
+        macaddr = path.split("/")[-1].replace("dev_", '').replace("_", ":")
         if self.on_disconnect is not None:
             if 'Connected' in changed:
                 if not changed['Connected']:
-                    self.on_disconnect()
+                    self.on_disconnect(macaddr)
+        else:
+            self.on_connect(macaddr)
```

Comment:
The logic here would allow `on_connect` to be called even if it was set to `None`. This would be a bad thing as most people I suspect will have it set to `None`.
Also, if the `on_disconnect` callback is changed to use a device object then this should also.

Reply:
my mistake , you are right I'll check it. Moreover i guess  i found another point where it misses the check , I'll let u know

Reply:
done and fixed in another package (peripheral).